pupil_src/player/player_methods.py

```diff
@@ -200,25 +211,61 @@ def update_recording_v087_to_v091(rec_dir):
 
     update_meta_info(rec_dir,meta_info)
 
+
 def update_recording_v091_to_v093(rec_dir):
     logger.info("Updating recording from v0.9.1 format to v0.9.3 format")
     meta_info_path = os.path.join(rec_dir,"info.csv")
-    pupil_data = load_object(os.path.join(rec_dir, "pupil_data"))
-
+    pupil_data = load_legacy_object(os.path.join(rec_dir, "pupil_data"))
 
     for g in pupil_data.get('gaze_positions', []):
         # fixing recordings made with bug https://github.com/pupil-labs/pupil/issues/598
         g['norm_pos'] = float(g['norm_pos'][0]), float(g['norm_pos'][1])
 
-    save_object(pupil_data,os.path.join(rec_dir, "pupil_data"))
-
+    save_legacy_object(pupil_data, os.path.join(rec_dir, "pupil_data"))
 
-    with open(meta_info_path,'r',encoding='utf-8') as csvfile:
+    with open(meta_info_path, 'r', encoding='utf-8') as csvfile:
         meta_info = csv_utils.read_key_value_file(csvfile)
         meta_info['Data Format Version'] = 'v0.9.3'
-    update_meta_info(rec_dir,meta_info)
+    update_meta_info(rec_dir, meta_info)
 
 
+def update_recording_v093_to_v094(rec_dir):
+    logger.info("Updating recording from v0.9.3 to v0.9.4.")
+    meta_info_path = os.path.join(rec_dir, "info.csv")
+
+    for file in os.listdir(rec_dir):
+        if file.startswith('.') or os.path.splitext(file)[1] in ('.mp4', '.avi'):
+            continue
+        rec_file = os.path.join(rec_dir, file)
+
+        def convert(data):
+            if isinstance(data, np.ndarray):
+                return data.tolist()
+            elif isinstance(data, str):
+                return data
+            elif isinstance(data, collections.Mapping):
```

Comment:
So that cases like `{key: <np.ndarray>}` or `[<np.ndarray>]` are also converted. `str` is a special case because `str([*'abc'])` results in `"['a', 'b', 'c']"`.